app/src/main/java/com/example/weatherapp/data/remote/NetworkResultExtensions.kt

```diff
@@ -64,7 +63,7 @@ fun <T> Flow<T>.mapToResult(): Flow<Result<T>> {
 
 suspend inline fun <T> Flow<Result<T>>.collectResult(
     crossinline onSuccess: (T) -> Unit,
-    crossinline onError: (Result.Error<T>) -> Unit,
+    crossinline onError: suspend (Result.Error<T>) -> Unit,
```

Comment:
Don't use suspend function here. Instead use normal function and then scope the work you are doing in the place that is invoking it